core_planning/twist_gate/test/src/test_twist_gate.cpp

```diff
@@ -15,63 +15,64 @@
  */
 
 #include <gtest/gtest.h>
-#include <ros/ros.h>
-
 #include "test_twist_gate.hpp"
 
+
 class TwistGateTestSuite : public ::testing::Test
 {
 public:
   TwistGateTestSuite() {}
   ~TwistGateTestSuite() {}
 
-  TwistGateTestClass test_obj_;
+  std::shared_ptr<TwistGateTestClass> test_obj_;
 
 protected:
   virtual void SetUp()
   {
-    ros::NodeHandle nh;
-    ros::NodeHandle private_nh("~");
-    test_obj_.tg = new TwistGate(nh, private_nh);
+    test_obj_ = std::make_shared<TwistGateTestClass>();
+    
   };
-  virtual void TearDown() { delete test_obj_.tg; }
+  virtual void TearDown() { test_obj_->tg.reset();}
 };
 
 TEST_F(TwistGateTestSuite, twistCmdCallback)
 {
   double linear_x = 5.0;
   double angular_z = 1.5;
-  test_obj_.publishTwistCmd(linear_x, angular_z);
-  ros::WallDuration(0.1).sleep();
-  test_obj_.tgSpinOnce();
+  
+  test_obj_->publishTwistCmd(linear_x, angular_z);
+  rclcpp::sleep_for(std::chrono::milliseconds(100));
+  rclcpp::spin_some(test_obj_->tg->get_node_base_interface());
+
   for (int i = 0; i < 3; i++)
   {
-    ros::WallDuration(0.1).sleep();
-    ros::spinOnce();
+    rclcpp::sleep_for(std::chrono::milliseconds(100));
+    rclcpp::spin_some(test_obj_->get_node_base_interface());
+    
   }
-
-  ASSERT_EQ(linear_x, test_obj_.cb_vehicle_cmd.twist_cmd.twist.linear.x);
-  ASSERT_EQ(angular_z, test_obj_.cb_vehicle_cmd.twist_cmd.twist.angular.z);
+  // TODO: These tests need use_twist to be set to true
```

Comment:
TODO?

Reply:
I added this comment here while unit testing. It seems like both even in the earlier implementation these unit tests were failing without the required change set. The TODO is because the unit tests are commented out with the required fix in the comment. Should i just fix it here?

Reply:
We target unit test parity with ROS2, so this is fine.